netCDF4/utils.py

```diff
@@ -305,10 +305,10 @@ def _StartCountStride(elem, shape, dimensions=None, grp=None, datashape=None,\
                 ee = range(start,stop,step)
             except ValueError: # start, stop or step is not valid for a range
                 ee = False
-            if no_get_vars and ee and len(e) == len(ee) and (e == np.arange(start,stop,step)).all():
+            if ee and len(e) == len(ee) and (e == np.arange(start,stop,step)).all():
                 # don't convert to slice unless abs(stride) == 1
                 # (nc_get_vars is very slow, issue #680)
-                if step not in [1,-1]:
+                if no_get_vars and step not in [1,-1]:
```

Comment:
For netcdf-c prior to 4.6.2, we want to avoid using strided access, so if `no_get_vars=True` *and* the slice step is not 1 we don't convert to a slice.

Reply:
Yep, that makes sense. (Side, it's usually best to avoid names like `no_something` so you don't end up with double-negatives like `not no_something`)

Reply:
Good idea, I've changed `no_get_vars` to `use_get_vars` for clarity.